dashboard/db/migrate/20220408173143_fix_projects_index.rb

```diff
@@ -0,0 +1,9 @@
+class FixProjectsIndex < ActiveRecord::Migration[6.0]
+  def change
+    existing_index = indexes('projects').select {|i| i.name == "storage_apps_project_type_index"}.first
+    if existing_index.lengths.blank?
+      remove_index :projects, :project_type
+      add_index :projects, :project_type, length: 191, name: 'storage_apps_project_type_index'
+    end
+  end
```

Comment:
nice solution! two thoughts:
1. I believe your logic is correct, but I would still feel more comfortable if this migration was explicitly prevented from running in production. removing the index for any period of time in production could be really bad.
2. as another precaution, I would suggest putting all this in the `up` method rather than the `change` method, and (if necessary) adding an empty `down` method. if the migration fails for any reason, I don't think we'll want to run the same code again when trying to roll it back. (this feels more optional)